fetch-utils.js

```diff
@@ -4,27 +4,37 @@ const SUPABASE_KEY =
 const client = supabase.createClient(SUPABASE_URL, SUPABASE_KEY);
 
 export async function getCountries(name, continent) {
-    // > Part D: Add a second argument to `select(` to 
-    // return an exact db count of matching records
-
     // > Part A: Implement the client query from countries:
-    //   1. select all columns
-    //   2. order by country name
-    //   3. limit to 100 countries
-    let query = // ?
+    let query = client
+        .from('countries')
+        //   1. select all columns
+        // > Part D: Add a second argument to `select(` to
+        // return an exact db count of matching records
+        .select('*', { count: 'exact' })
+        //   2. order by country name
+        .order('name')
+        //   3. limit to 100 countries
+        .limit(100);
 
     if (name) {
         // > Part C: add query for name
+        query = query.ilike('name', `%${name}`);
     }
-    
+
     if (continent) {
         // > Part C: add query for continent
+        query = query.eq('continent', continent);
     }
```

Comment:
you'll see this paradigm a lot in backend — taking a response (or request), passing it to a function, having that function do something with the request, then that function passes it along